Functions meetnetten

ToonHub opened the pull request:
I would like to add some functions to read data from the meetnetten-database.
I also added a vignette.

ToonHub committed:
add function get_meetnetten_visits()

ToonHub committed:
add function get_meetnetten_observations
update function get_meetnetten_visits

ToonHub committed:
add functions get_meetnetten_schemes and get_meetnetten_locations

fredericpiesschaert committed:
Initial commit of function get_taxonlijsten_lists

fredericpiesschaert committed:
initial commit of function get_taxonlijsten_features

fredericpiesschaert committed:
Initial commit of function get_taxonlijsten_items

fredericpiesschaert committed:
minor change in function description

fredericpiesschaert committed:
syntax cleaning

ToonHub committed:
meetnetten functions: update documentation

ToonHub committed:
updat DESCRIPTION file

ToonHub committed:
meetnetten functions: add vignette

fredericpiesschaert committed:
initial commit taxonlijsten vignette

fredericpiesschaert committed:
add tidyverse requirement

fredericpiesschaert committed:
roxygenize

fredericpiesschaert committed:
some issues from build tests

fredericpiesschaert committed:
add missing libraries

fredericpiesschaert committed:
Merge branch 'main' into functions_taxonlijsten

# Conflicts:
#	vignettes/get_data_inboveg_results.Rda

fredericpiesschaert committed:
update rda

fredericpiesschaert committed:
cleaning things up

fredericpiesschaert committed:
update description file and release notes

fredericpiesschaert committed:
Update R/get_taxonlijsten_items.R

Co-authored-by: Els Lommelen <els.lommelen@inbo.be>

fredericpiesschaert committed:
code optimisation

fredericpiesschaert committed:
cleaning code

ToonHub committed:
get_meetnetten_observations: update documentation

ToonHub committed:
finish vignette for meetnetten functions

ToonHub committed:
add .rda file for meetnetten vignette

ToonHub commented:
Here you find the rendered vignette in which the meetnetten functions are explained.

[get_data_meetnetten.zip](https://github.com/user-attachments/files/15969068/get_data_meetnetten.zip)

ElsLommelen committed:
update names of (earlier) vignettes

fredericpiesschaert committed:
Merge pull request #57 from inbo/quickfix

update names of (earlier) vignettes

ElsLommelen commented:
Bedankt!  Terwijl je aan je branch bezig was, is er een nieuwe versie van inbodb uitgekomen, waardoor je branch conflicten geeft met de huidige main (en problemen bij de checks).  Zou je je branch even kunnen updaten naar deze nieuwste versie en de conflicten oplossen (zie eerste puntje [hier](https://inbo.github.io/git-course/workflow_review.html#Merge_changes_to_main))?
Dit geeft normaal nog 2 errors die je met de aanpassingen uit commit d4ce630 zou kunnen oplossen, moest je dit handig vinden.  (En met `checklist::check_package()` kan je dan eventueel lokaal testen of er verder nog problemen zijn, mijn review zal voor morgen zijn.)

Momenteel is @fredericpiesschaert ook bezig met nieuwe functies en een vignette, dus misschien kunnen we die na afwerken samenvoegen in 1 release?

fredericpiesschaert committed:
keep tweaking

ThierryO commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+              msg = "Not a connection object to database.")
+
+  if (!is.null(scheme_name)) {
+    assert_that(is.character(scheme_name))
```
Unless you want to select multiple schemas use this. 
```suggestion
    assert_that(is.string(scheme_name), noNA(schema_name))
```

Otherwise use
```suggestion
    assert_that(is.character(scheme_name), length(scheme_name) > 0, noNA(scheme_name))
```

ThierryO commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+
+  query_result <- tbl(connection, sql(sql_statement))
+
+  if (!is.null(scheme_name) & !is.null(species_group_selected)) {
```
Always use '&&' in if statements
```suggestion
  if (!is.null(scheme_name) && !is.null(species_group_selected)) {
```

ThierryO commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+      filter(str_to_lower(.data$scheme) %in% scheme_name |
+               str_to_lower(.data$species_group) %in% species_group_selected)
+
+  } else if (!is.null(scheme_name)) {
```
write the structure as as nested if

```
if (!is.null(schema_name)) {
    if (!is.null(species_group) {
    } else {
    }
else
```

ToonHub replied:
OK aangepast

ThierryO commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+#' = \code{FALSE}), which means that no observations can be recorded any more.
+#'
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter arrange collect tbl sql mutate select
```
import all functions. You're missing %>%, .data, ...

ToonHub replied:
OK

ThierryO commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+
+  if (!is.null(scheme_name)) {
+    assert_that(is.character(scheme_name))
+    scheme_name <- str_to_lower(scheme_name)
```
Use a base function to reduce the number of dependencies.

```suggestion
    scheme_name <- tolower(scheme_name)
```

ThierryO commented on R/get_meetnetten_schemes.R:
```diff
@@ -0,0 +1,71 @@
+    INNER JOIN staging_meetnetten.projects_projectgroup pg ON pg.id = p.group_id
+    INNER JOIN staging_meetnetten.projects_project_protocols ppp ON ppp.project_id = p.id
+    INNER JOIN staging_meetnetten.protocols_protocol PR ON PR.id = ppp.protocol_id
+    WHERE 1 = 1"
```
`WHERE 1=1` is redundant and do the sorting in the database.

```suggestion
    ORDER BY pg.name, p.name, pr.name"
```

ThierryO commented on R/get_meetnetten_schemes.R:
```diff
@@ -0,0 +1,71 @@
+    INNER JOIN staging_meetnetten.protocols_protocol PR ON PR.id = ppp.protocol_id
+    WHERE 1 = 1"
+
+  query_result <- tbl(connection, sql(sql_statement)) %>%
```
simply use this if you do the sorting in the database

```suggestion
  query_result <- dbGetQuery(connection, sql_statement)
```

ThierryO commented on vignettes/get_data_meetnetten.Rmd:
````diff
@@ -0,0 +1,296 @@
+rm(con)
+```
+
+```{r echo=FALSE, include=FALSE}
````
only save the image when the file in not present

ElsLommelen commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter arrange collect tbl sql mutate select
+#' @importFrom stringr str_to_lower
+#' @importFrom sf st_as_sf st_drop_geometry
```
Idee van inbodb is dat het een licht package blijft met niet te veel dependencies, zodat niemand tegengehouden wordt om het te installeren of als dependency te gebruiken.  Dat er enkele data(bank)gerelateerde dependencies zijn, is logisch, maar een package als sf vind ik toch een beetje 'out-of-scope' voor iemand die bv. florabank wil bevragen of enkel de `connect_inbo_db()`-functie nodig heeft.  (Dit hangt behalve R-packages ook af van software als 'geos' en 'gdal'.)  Zou je dit kunnen ondervangen door in deze functie `requireNamespace()` te gebruiken zoals [hier](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Suggested-packages-1) uitgelegd, zodat package sf naar suggests kan en niet standaard geïnstalleerd wordt bij alle gebruikers van inbodb?

ToonHub replied:
OK. Is aangepast.

ElsLommelen commented on R/get_meetnetten_observations.R:
```diff
@@ -0,0 +1,238 @@
+#' con <- connect_inbo_dbase("S0008_00_Meetnetten")
+#'
+#' # get observations for a specific monitoring scheme and collect data
+#' observations_treefrog <- get_meetnetten_observation(con,
```
```suggestion
#' observations_treefrog <- get_meetnetten_observations(con,
```

ElsLommelen replied:
En ook hier zou ik de output van de voorbeelden gewoon weergeven en niet wegschrijven in een variabele, dat is niet zo handig voor een gebruiker die ze even test.

ElsLommelen commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+#'                                                  scheme_name = "Heivlinder")
+#'
+#' main_locations <- locations_heivlinder$main_locations
+#' sublocations <- locations_heivlinder$sublocations
```
```suggestion
#' locations_heivlinder$main_locations
#' locations_heivlinder$sublocations
```
Als gebruiker van een package vind ik het eigenlijk handiger als voorbeelden direct een output geven en geen variabelen opslaan in mijn workspace, dus ik zou het zo aanpassen.  Misschien overwegen om dit ook bij de andere voorbeelden te doen?

ElsLommelen commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+#'
+#' # Close the connection when done
+#' dbDisconnect(con)
+#' rm(con)
```
Samen met de connectie misschien ook de aangemaakte variabelen verwijderen?  (Moesten de voorbeelden niet binnen een `dontrun` gestaan hebben, dan was dit sowieso iets waar je een opmerking over zou krijgen.)

ThierryO replied:
Dat is niet echt nodig.

ElsLommelen commented on vignettes/get_data_meetnetten.Rmd:
```diff
@@ -0,0 +1,296 @@
+
+## Information on monitoring schemes
+
+The function `get_meetnetten_schemes` gives an overview per species group of all
```
```suggestion
The function `get_meetnetten_schemes()` gives an overview per species group of all
```
Als je een functienaam laat volgen door haakjes, dan maakt pkgdown op de website automatisch een link naar de documentatie van de functie in kwestie, wat voor een gebruiker heel handig is (naast het feit dat hij/zij zo impliciet sneller door heeft dat het over een functie gaat).  Misschien handig om dit overal in de documentatie toe te passen bij verwijzingen naar functies (ook bij de help van functies en NEWS.md kan dit)?

(Zie ook [hier](https://pkgdown.r-lib.org/articles/linking.html).)

ToonHub replied:
OK. Is aangepast.

ElsLommelen commented on vignettes/get_data_meetnetten.Rmd:
```diff
@@ -0,0 +1,296 @@
+
++ `is_active`. When `is_active` = `FALSE`, the location is not counted any more.
+This is the case when the location appears to be inaccessible or when the target
+species does not occur at the location any more.
```
```suggestion
species does not occur at the location anymore.
```

ElsLommelen commented on vignettes/get_data_meetnetten.Rmd:
```diff
@@ -0,0 +1,296 @@
+
+Next, let's have a look at the sublocations of the location Arenbergpolder.
+The locations consists of 14 sections.
+Note that we only show the active (`is_active` = `TURE`) sublocations. 
```
```suggestion
Note that we only show the active (`is_active` = `TRUE`) sublocations. 
```

ElsLommelen commented on vignettes/get_data_meetnetten.Rmd:
```diff
@@ -0,0 +1,296 @@
+Since many of the target species are sensitive species, we do not publicly make 
+available the exact location of the observations.
+Location information is generalized to 1, 5 or 10 km Universal Transverse 
+Mercator (UTM) grid cells, when publishing the data on [GBIF](www.gbif.org).
```
```suggestion
Mercator (UTM) grid cells, when publishing the data on [GBIF](https://www.gbif.org).
```
`https://` toevoegen is nodig om bij klikken op een link op een website uit te komen (redirect werkt enkel als je een website zelf intypt in een browser)

ElsLommelen commented on vignettes/get_data_meetnetten.Rmd:
```diff
@@ -0,0 +1,296 @@
+Location information is generalized to 1, 5 or 10 km Universal Transverse 
+Mercator (UTM) grid cells, when publishing the data on [GBIF](www.gbif.org).
+The generalisation rules are the same as in 
+[waarnemingen.be](www.waarnemingen.be).
```
```suggestion
[waarnemingen.be](https://www.waarnemingen.be).
```

ElsLommelen commented on vignettes/get_data_meetnetten.Rmd:
```diff
@@ -0,0 +1,296 @@
+
+Please respect this policy when you publish results based on the meetnetten
+data.
+When in doubt, contact [info@meetnetten.be](info@meetnetten.be).
```
```suggestion
When in doubt, contact <info@meetnetten.be>.
```
Ik vermoed dat dit het gedrag geeft wat je zou willen?  (De link gaf in elk geval aan dat het bestand waarnaar verwezen werd, niet bestond.)
Eventueel ook even nagaan in hoeverre het een goed idee is om dat emailadres open op het internet te zetten.  Er zijn manieren om dit te escapen (iets rond het emailadres zetten), waardoor het moeilijker te lezen is voor spambots, al weet ik niet of dat nog gangbaar is.

ElsLommelen commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+#' @param scheme_name the name of the monitoring scheme for which you want to
+#' extract location data. Data from multiple schemes can be selected by providing
+#' a vector with the names of the schemes.
+#' @param species_group the name of the species group for which you want to
+#' extract location data. Data from multiple species groups can be selected by
+#' providing a vector with the names of the species groups.
```
Het zou nuttig kunnen zijn om hier de default NULL toch op een of andere manier te vermelden.  Als ik de description van hierboven niet (met al te veel aandacht) gelezen heb en ik niet besef dat ik dit niet moet invullen, dan kan dit verwarrend zijn (NULL om alles terug te krijgen is ook een beetje tegen de intuïtie).  En ik kan me best wel wat gebruikers voorstellen die gewoon alle argumenten proberen in te vullen, en dan botsen tegen de moeilijkheid dat ze niet weten wat ze moeten doen als ze alles willen.  (Dat staat ook niet echt als voorbeeld vermeld.)
Dit mag gerust een impliciete vermelding zijn, bv. 'If not all observations of all monitoring schemes are wanted, add here the name of the species group for...'
(idem for scheme_name)

ElsLommelen commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+#'
+#' @details
+#' Each monitoring scheme of the species monitoring programme of Flanders
+#' \href{www.meetnetten.be}{Meetnetten} consists of a fixed set of locations.
```
Deze link geeft de foutmelding 'kan het bestand niet vinden', vermoedelijk omdat er geen `https://` voor staat?

ThierryO replied:
iid. URL's moet je volledig geven. Anders is dat een relatief path.

ElsLommelen commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+#' A monitoring scheme for rare species includes all locations where the species
+#' occurs. For more common species a sample of locations is drawn and the
+#' the selected locations are included in the monitoring scheme. In some cases,
+#' the monitoring project in \href{www.meetnetten.be}{Meetnetten} also contains
```
Idem (en best alle linken nakijken)

ThierryO replied:
Rcmdcheck via checklist zou dat in principe moeten detecteren.

ElsLommelen commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+#'                                                   species_group = "libellen")
+#'
+#' main_locations <- locations_dragonflies$main_locations
+#' sublocations <- locations_dragonflies$sublocations
```
Dit geeft een lege tabel, evt. aangeven dat dit de bedoeling is?

fredericpiesschaert committed:
set collect = TRUE for chunks with output in vignette

ElsLommelen commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+  locations_sublocations <- locations_sf %>%
+    filter(.data$location_type == "sublocation") %>%
+    rename(sublocation = location, location_id = parent_id) %>%
+    left_join(locations_main_names, by = c("scheme", "location_id")) %>%
```
left_join wordt niet geïmporteerd

ElsLommelen commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+    filter(!is.na(.data$location))
+
+  locations_main <- locations_main %>%
+    select(species_group, scheme, location, is_sample, is_active)
```
```suggestion
    select("species_group", "scheme", "location", "is_sample", "is_active")
```
en check ook andere opmerkingen die `lintr::lint_package()` geeft

ElsLommelen commented on R/get_meetnetten_observations.R:
```diff
@@ -0,0 +1,238 @@
+#' @param species_group the name of the species group for which you want to
+#' extract visit data. Data from multiple species groups can be selected by
+#' providing a vector with the names of the species groups.
+#' @param connection dbconnection with the database 'S0008_00_Meetnetten'
+#' on the inbo-sql08-prd.inbo.be server
```
Vermits dit dezelfde omschrijving is als voor functie `get_meetnetten_locaties()`, kan je dit ook vervangen door `@inheritParams get_meetnetten_locaties`.  Het grote voordeel zit in het onderhoud achteraf: als je een aanpassing wil doen, moet je dat maar eenmaal doen.  (Uiteraard is het voordeel hier beperkt omdat het maar over 1 argument gaat, zeker als je meerdere argumenten dezelfde omschrijving zou geven, is dat heel handig.)

ElsLommelen commented on R/get_meetnetten_observations.R:
```diff
@@ -0,0 +1,238 @@
+#' monitoring schemes are returned.
+#'
+#' @param scheme_name the name of the monitoring scheme for which you want to
+#' extract visit data. Data from multiple schemes can be selected by providing
```
Zijn observations ook 'visist data'?  Ook bij species_group hieronder wordt verwezen naar visit data.

ElsLommelen commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,178 @@
+#' @param connection dbconnection with the database 'S0008_00_Meetnetten'
+#' on the inbo-sql08-prd.inbo.be server
+#'
+#' @return A list with two sf objects:
```
Is er een reden waarom deze functie geen optie `collect = FALSE` heeft die de gebruiker toelaat om eerst de gegevens te bekijken?  Zou het mogelijk zijn om zoiets aan te bieden, evt. zonder de omzetting naar sf?  (Dan zou je in een voorbeeld kunnen aangeven hoe dit moet gebeuren, zodat de gebruiker dit zelf kan doen na eventuele datamanipulaties op de databanktabel.)  Om eens over na te denken...

fredericpiesschaert committed:
keep trying

ElsLommelen commented on R/get_meetnetten_observations.R:
```diff
@@ -0,0 +1,238 @@
+#'    \item \code{target_species}: \code{TRUE} when the observed species is the target
+#'    species, \code{FALSE} when the observed species is a secondary species
+#'    (another species than the target species that can be counted with the
+#'    same protocol,, see details)
```
```suggestion
#'    same protocol, see details)
```

ElsLommelen commented on R/get_meetnetten_observations.R:
```diff
@@ -0,0 +1,238 @@
+#'    \item \code{count_type}: most of the time the number of individuals are
+#'    counted (\code{count_type} = \code{exact count}), however for some
+#'    monitoring schemes different type of counts are performed. Check the
+#'    protocol for more information when this is the case.
```
Eventueel link toevoegen naar waar die protocollen te vinden zijn?

ElsLommelen commented on R/get_meetnetten_observations.R:
```diff
@@ -0,0 +1,238 @@
+#' @importFrom stringr str_to_lower
+#'
+#' @details The species monitoring programme of Flanders
+#' (\href{www.meetnetten.be}{Meetnetten}) consists of a series of monitoring
```
Link werkt niet in html (zie gelijkaardige commentaren)

ToonHub replied:
OK. Is aangepast.

ElsLommelen commented on R/get_meetnetten_observations.R:
```diff
@@ -0,0 +1,238 @@
+  if (!is.null(scheme_name) & !is.null(species_group_selected)) {
+
+    query_result <- query_result %>%
+      filter(str_to_lower(.data$scheme) %in% scheme_name |
```
Ook hier ontbreken minstens de imports voor `%>%` en `.data`, dus best alle gebruikte functies nog eens even nakijken.

ElsLommelen commented on R/get_meetnetten_visits.R:
```diff
@@ -0,0 +1,169 @@
+#'    \item \code{protocol}: the protocol used
+#'    \item \code{location}: the name of the location
+#'    \item \code{visit_id}: unique id for a count event
+#'    \item \code{validation_status}: validation status of the visit (visits
+#'    that are not approved are not provided)
+#'    \itemize{
+#'        \item \code{10}: visit not validated
+#'        \item \code{100}: visit approved
+#'        }
```
Euh, wat is 'validated' en wat is 'approved'?  Ik zou denken dat een 'visit' dat 'not validated' is, zeker niet approved is?

Na even nadenken denk ik dat ik hem door heb, misschien er dit van maken?
```suggestion
#'    \item \code{validation_status}: validation status of the visit (visits
#'    that are validated but not approved are not provided)
#'    \itemize{
#'        \item \code{10}: visit not validated
#'        \item \code{100}: visit approved
#'        }
```

fredericpiesschaert replied:
valideren is het proces, goed/afkeuren is het resultaat van dat proces

ElsLommelen reviewed (commented):
Ik heb de volledige teksten en code doorgenomen en de grote lijnen zien er goed uit (duidelijke omschrijvingen, overzichtelijke en werkende code,...).  Ik heb een aantal aandachtspunten aangegeven, maar ik ben deze niet consequent blijven herhalen voor elk van de functies of doorheen het hele vignet.  Het gaat vaak over details als niet werkende links, niet geïmporteerde functies, een suggestie bij de voorbeelden,... dus kijk best even na of de opmerkingen ook op andere plaatsen van toepassing zijn (ook die van @ThierryO trouwens).  Idealiter zouden de checks ook groene vinkjes geven, maar zodra je de laatste versie van de main binnengehaald hebt, zou die GHA zelf terug in orde moeten zijn, waardoor je hiermee aan de slag kunt gaan.

(Je kan alles trouwens ook lokaal testen met `checklist::check_package()`.)

fredericpiesschaert committed:
remove collect = TRUE from dbGetQuery

fredericpiesschaert committed:
replace underscore by hyphen in image-names (checklist suggestion)

fredericpiesschaert committed:
Update vignettes/get_data_taxonlijsten.Rmd

Co-authored-by: Els Lommelen <els.lommelen@inbo.be>

fredericpiesschaert commented on vignettes/get_data_meetnetten.Rmd:
```diff
@@ -0,0 +1,296 @@
+# Introduction
+
+The Flemish species monitoring programme [meetnetten.be](www.meetnetten.be) 
+consists of set of monitoring schemes for priority species in Flanders, the 
```
```suggestion
consists of a set of monitoring schemes for priority species in Flanders, the 
```

fredericpiesschaert commented on vignettes/get_data_meetnetten.Rmd:
```diff
@@ -0,0 +1,296 @@
+
+These functions will only work for people with access to the INBO network. 
+As an INBO employee, you should make sure you have reading-rights for the INBO 
+DWH Server, otherwise place an ICT-call.
```
```suggestion
Datawarehouse server, otherwise place an ICT-call.
```

fredericpiesschaert commented on vignettes/get_data_meetnetten.Rmd:
````diff
@@ -0,0 +1,296 @@
+```
+
+These functions will only work for people with access to the INBO network. 
+As an INBO employee, you should make sure you have reading-rights for the INBO 
````
```suggestion
As an INBO employee, you should make sure you have read rights for the INBO 
```

fredericpiesschaert commented on vignettes/get_data_meetnetten.Rmd:
```diff
@@ -0,0 +1,296 @@
+We present some functions to easily access the data from the meetnetten 
+database.
+
+# Aim
```
Should we mention that there is a time delay between the production database and the datawarehouse?

fredericpiesschaert commented on vignettes/get_data_meetnetten.Rmd:
````diff
@@ -0,0 +1,296 @@
+main_locations %>%
+  head(5)
+```
+Two variables need some further explanations:
````
```suggestion
Two variables need some further explanation:
```

fredericpiesschaert commented on vignettes/get_data_meetnetten.Rmd:
```diff
@@ -0,0 +1,296 @@
+Locations that are not selected, indicated by `is_sample` = `FALSE`, can be
+counted optionally.
+
++ `is_active`. When `is_active` = `FALSE`, the location is not counted any more.
```
```suggestion
+ `is_active`. When `is_active` = `FALSE`, the location is no longer counted.
```

ElsLommelen commented:
@ToonHub Het lijkt me het eenvoudigst als jij je branch meteen rebaset naar branch `functions_taxonlijsten` van @fredericpiesschaert (die intussen volledig afgewerkt is).  Je mag deze PR naar de main gewoon behouden, dan mergen we de PR van @fredericpiesschaert niet rechtstreeks naar de main (maar laat die PR voorlopig maar open staan, die verdwijnt normaal vanzelf als de inhoud via uw branch gemerged wordt met de main).
Je mag dan het versienummer van Frederic behouden en in `NEWS.md` een korte beschrijving van je aanpassingen toevoegen.

fredericpiesschaert commented on vignettes/get_data_meetnetten.Rmd:
````diff
@@ -0,0 +1,296 @@
+  select(location, sublocation, start_date, 
+         scientific_name, count)
+```
+Optionally observers can also record other species that can be counted using the
````
```suggestion
Optionally observers can also record non target species that can be counted using the
```

fredericpiesschaert commented on vignettes/get_data_meetnetten.Rmd:
````diff
@@ -0,0 +1,296 @@
+save.image(file = "get_data_meetnetten_results.Rda")
+```
+
+# Data policy
````
Wat met projecten met beperkte toegang, zoals de wintertellingen vleermuizen? Moeten we niet meegeven dat die vleermuizendata zeker nooit zomaar gepubliceerd mogen worden zonder overleg met de vleermuiswerkgroep?

ElsLommelen replied:
Suggestie: als je zeker wil zijn dat iedereen die boodschap krijgt, overweeg dan om (ook) een warning te geven ingeval die gegevens opgevraagd worden.  Ga er zeker niet van uit dat iedereen het vignet leest en de volledige inhoud onthoudt tot het eens van toepassing is. ;-)

ToonHub replied:
In de tutorial houd ik het graag zo generiek mogelijk. Ik heb nu toegevoegd om altijd info@meetnetten.be te contacteren voor resultaten te publiceren. Zo kunnen we nagaan of de publicatie conform is met de afspraken die dan geldig zijn. Meer details kunnen we dan op het infomoment geven.

ElsLommelen replied:
Lees jij altijd een tutorial vooraleer je een package gebruikt?  Eerlijk gezegd: als je het echt belangrijk vindt dat gebruikers die boodschap krijgen, dan zou ik die boodschap toch ergens geven bij het runnen van de functie(s).  Dit moet geen warning zijn, het mag gewoon een 'message' zijn die verschijnt tijdens het runnen van een `get_meetnetten`-functie.  (Doordat het maar over een deel van de functies van inbodb gaat, lijkt een opstartboodschap bij inbodb me minder geschikt, anders was dat ideaal geweest.)  Tidyverse heeft soms boodschappen die eenmaal om de 8 uur verschijnen, misschien is zoiets een ideetje om de gebruiker niet te vervelen met constante boodschappen en ze er toch bij elk gebruik even aan te herinneren?

fredericpiesschaert commented on R/get_meetnetten_observations.R:
```diff
@@ -0,0 +1,238 @@
+#' It is also important to know that counts can be recorded in the
+#' \href{www.meetnetten.be}{Meetnetten} website or by using the Meetnetten app.
+#' When using the Meetnetten app, the GPS coordinates of all observations are
+#' recorded and the observations are assigned to a locations or sublocation
```
```suggestion
#' recorded and the observations are assigned to a location or sublocation
```

ToonHub committed:
Update R/get_meetnetten_locations.R

Co-authored-by: Thierry Onkelinx <ThierryO@users.noreply.github.com>

ToonHub committed:
Update R/get_meetnetten_locations.R

Co-authored-by: Thierry Onkelinx <ThierryO@users.noreply.github.com>

ToonHub committed:
Update R/get_meetnetten_locations.R

Co-authored-by: Thierry Onkelinx <ThierryO@users.noreply.github.com>

ToonHub committed:
Update R/get_meetnetten_schemes.R

Co-authored-by: Thierry Onkelinx <ThierryO@users.noreply.github.com>

ToonHub committed:
Update R/get_meetnetten_schemes.R

Co-authored-by: Thierry Onkelinx <ThierryO@users.noreply.github.com>

ToonHub committed:
Update R/get_meetnetten_observations.R

Co-authored-by: Els Lommelen <els.lommelen@inbo.be>

ToonHub committed:
Update R/get_meetnetten_locations.R

Co-authored-by: Els Lommelen <els.lommelen@inbo.be>

ToonHub committed:
Update vignettes/get_data_meetnetten.Rmd

Co-authored-by: Frederic Piesschaert <frederic.piesschaert@inbo.be>

ToonHub committed:
Update vignettes/get_data_meetnetten.Rmd

Co-authored-by: Frederic Piesschaert <frederic.piesschaert@inbo.be>

ToonHub committed:
Update vignettes/get_data_meetnetten.Rmd

Co-authored-by: Frederic Piesschaert <frederic.piesschaert@inbo.be>

ToonHub committed:
Update vignettes/get_data_meetnetten.Rmd

Co-authored-by: Frederic Piesschaert <frederic.piesschaert@inbo.be>

ToonHub committed:
Update vignettes/get_data_meetnetten.Rmd

Co-authored-by: Frederic Piesschaert <frederic.piesschaert@inbo.be>

ToonHub committed:
Update R/get_meetnetten_observations.R

Co-authored-by: Frederic Piesschaert <frederic.piesschaert@inbo.be>

ToonHub committed:
Update vignettes/get_data_meetnetten.Rmd

Co-authored-by: Frederic Piesschaert <frederic.piesschaert@inbo.be>

ToonHub committed:
use tolower instead of str_to_lower
asserthat noNA scheme_name species_group
modify if else statement to make it nested
add missing functions to @importFrom

ToonHub committed:
remove variables in examples

ToonHub committed:
add () after function names in text

ToonHub committed:
correct hyperlinks

ToonHub committed:
modify examples
'validated and approved' instead of validated

ToonHub committed:
add sf to suggested packages and use requireNamespace("sf") in get_meetnetten_locations()

ToonHub committed:
mention project partners
modify data policy part

ToonHub committed:
update namespace

ToonHub committed:
correct error in if else statement

ToonHub committed:
update documentation

ThierryO commented:
@ToonHub graag dringend rebasen naar de branch van Frederik. Dat lost het probleem met de falende checklist test op.

ThierryO commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,204 @@
+#'
+#' @description This function queries the Meetnetten database for the locations
+#' and sublocations for a specified monitoring scheme or for all monitoring
+#' schemes within a specified species group. When no monitoring scheme or
```
Start zinnen telkens op een nieuwe lijn. Dat geeft overzichtelijker git diff wanneer je later dingen moet wijzigen.

ThierryO commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,204 @@
+#' @return When the \code{sf} package is installed, a list with two \code{sf}
+#' objects is returned:
+#'
+#' \itemize{
```
Gebruik markdown syntax. Dat is leesbaarder.

ThierryO commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,204 @@
+
+      query_result <- query_result %>%
+        filter(str_to_lower(.data$scheme) %in% scheme_name)
+      }
```
zorg voor correct gebruikt van witruimte. De } moet op dezelfde positie staan als de bijhorende if.

ThierryO commented on R/get_meetnetten_locations.R:
```diff
@@ -0,0 +1,204 @@
+      }
+    }else {
+
+      if (!is.null(species_group_selected)) {
```
voeg samen met else to elseif

ElsLommelen commented on DESCRIPTION:
```diff
@@ -10,6 +10,11 @@ Authors@R: c(
            comment = c(ORCID = "0000-0002-6378-6229", affiliation = "Research Institute for Nature and Forest (INBO)")),
+    person("Frederic", "Piesschaert", , "frederic.piesschaert@inbo.be", role = "aut",
+           comment = c(ORCID = "0000-0002-5843-646X", affiliation = "Research Institute for Nature and Forest (INBO)")),
+           comment = c(ORCID = "0000-0002-6378-6229")),
```
```suggestion
```
Ik denk dat er hier iets misgelopen is met het rebasen, deze regel lijkt hier te veel te staan

ElsLommelen reviewed (commented):
Omdat ik zelf nog een klein issue moet oplossen (onderhoud package), wou ik uw branch binnenhalen, en daarbij heb ik even gecheckt of de aanpassingen van Frederic in uw branch terechtgekomen zijn.  En op dit klein foutje na is dat allemaal in orde.
(Je zal wel zelf nog even `checklist::check_package()` moeten runnen en de gegenereerde aanpassingen committen om uw naam in de Zenodo-file, citatie,... te krijgen.  En ook belangrijk: je eigen aanpassingen nog toevoegen aan NEWS.md.)

ElsLommelen committed:
remove odbcConnectActions (fixes #52)

ElsLommelen committed:
import dependency setMethod from methods (fixes comment in PR #51)

ToonHub committed:
Merge pull request #58 from inbo/add-fixes

maintenance fixes

ToonHub committed:
update citation

ToonHub committed:
add affiliation

ToonHub committed:
update NAMESPACE

ToonHub committed:
anounce new meetnetten functions

ToonHub committed:
correct spelling and style issues

ToonHub committed:
update Rda files

ToonHub committed:
add tibble as suggested package for function as_tibble

ToonHub committed:
add , in chunck header
update rda file

ToonHub committed:
update licence
set eval TRUE for setup chunck

ElsLommelen commented on R/get_meetnetten_observations.R:
```diff
@@ -0,0 +1,246 @@
+#' scheme. Reports of the Research Institute for Nature and Forest (INBO) 2023
+#' (15).
+#' Research Institute for Nature and Forest (INBO), Brussels.
+#' \doi{10.21436/inbor.93332112}.
```
Een beetje dom om `doi` en `inbor` in het woordenboek te zetten, misschien bij checklist een issue plaatsen met de vraag of dit bij de spellingscontrole genegeerd kan worden?

ElsLommelen reviewed (approved):
Ziet er goed uit, ik heb enkel nog een kleinigheidje toegevoegd om eventueel te overwegen.

ElsLommelen commented:
@ThierryO Zou jij dit nog even willen approven als je gevraagde aanpassingen in orde zijn?

ToonHub committed:
remove tibble as suggested package as tibble functions are available in dplyr package
